fastlane/lib/fastlane/actions/jira.rb

```diff
@@ -21,18 +25,39 @@ def self.run(params)
                     password: password
                   }
 
-        client = JIRA::Client.new(options)
-        issue = client.Issue.find(ticket_id)
-        comment = issue.comments.build
-        comment.save({ 'body' => comment_text })
+        begin
+          client = JIRA::Client.new(options)
+          issue = client.Issue.find(ticket_id)
+          comment = issue.comments.build
+          comment.save({ 'body' => comment_text })
+
+          # An exact representation of the JSON returned from the JIRA API
+          # https://github.com/sumoheavy/jira-ruby/blob/master/lib/jira/base.rb#L67
+          json_response = comment.attrs
+          if json_response.nil?
+            UI.error('Failed to add a comment on JIRA ticket')
+            return nil
+          else
+            Actions.lane_context[SharedValues::JIRA_JSON] = json_response
+            UI.success('Successfully added a comment on JIRA ticket')
+            return json_response
+          end
```

Comment:
I think we could raise an error here if response is nil so that the `fail_on_error` option is respected 😇 Thoughts?

```suggestion
          raise 'Failed to add a comment on JIRA ticket' if json_response.nil?

          Actions.lane_context[SharedValues::JIRA_JSON] = json_response
          UI.success('Successfully added a comment on JIRA ticket')
          return json_response
```